zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/SQLPPGenerator.scala

```diff
@@ -334,6 +354,13 @@ class SQLPPGenerator extends AsterixQueryGenerator {
       case _ => aggFuncExpr(typeImpl.getAggregateStr(aggr.func))
     }
   }
+
+  protected def append(queryBuilder: StringBuilder, queryStr: String): Unit = {
+    if (!queryStr.isEmpty) {
```

Comment:
If `queryStr` is empty, it means the corresponding statement is empty as well. For example, an empty filter:Seq[FilterStatement] or unnest:Seq[UnnestStatement].